raiden/tests/test_service.py

```diff
@@ -29,6 +29,32 @@ def test_ping(raiden_network):
     assert decoded.echo == sha3(ping.encode() + app1.raiden.address)
 
 
+@pytest.mark.timeout(3)
```

Comment:
That's right. With the bug as is right now, ping hangs and pytest-timeout kills it after 3 seconds. That is the failing condition. With the patch applied, the re-sends are disabled after 5 fails so the test will just check that 5 pings were (tried to) sent and will pass it that's the case.